cns/service/main.go

```diff
@@ -773,18 +779,81 @@ func InitializeMultiTenantController(ctx context.Context, httpRestService cns.HT
 	return nil
 }
 
-// initializeCRD state
-func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cnsconfig configuration.CNSConfig) error {
-	var requestController singletenantcontroller.RequestController
+type nodeNetworkConfigGetter interface {
+	Get(context.Context) (*v1alpha.NodeNetworkConfig, error)
+}
 
-	logger.Printf("[Azure CNS] Starting request controller")
+type ncStateReconciler interface {
+	ReconcileNCState(ncRequest *cns.CreateNetworkContainerRequest, podInfoByIP map[string]cns.PodInfo, scalar v1alpha.Scaler, spec v1alpha.NodeNetworkConfigSpec) error
+}
 
-	kubeConfig, err := kubecontroller.GetKubeConfig()
+// TODO(rbtr) where should this live??
+// InitCNS initializes cns by passing pods and a createnetworkcontainerrequest
+func initCNS(ctx context.Context, cli nodeNetworkConfigGetter, ncReconciler ncStateReconciler) error {
+	// Get nnc using direct client
+	nnc, err := cli.Get(ctx)
 	if err != nil {
-		logger.Errorf("[Azure CNS] Failed to get kubeconfig for request controller: %v", err)
+		// If the CRD is not defined, exit
+		if crd.IsNotDefined(err) {
+			logger.Errorf("CRD is not defined on cluster: %v", err)
+			os.Exit(1)
```

Comment:
let `main()` be the one to exit the process? maybe this logic belongs in caller, or above

Reply:
changed to just return the error, and main is already set up to exit there 👍🏼